### PR TITLE
Feature/rr 204 build header functions

### PR DESCRIPTION
**Pull request by CaraHill**
https://govtnz.atlassian.net/browse/RR-204

This PR implements the header as per the new designs.

Screen Shot 2019-05-14 at 1 37 10 PM

This includes a nav bar at the top that appears on all pages, and the header bar for just the admin/rebate_form edit and show pages. The former nav bar has been removed and routes updated accordingly. Tests have been updated accordingly. 

Please note:
- It is not quite picture perfect to the design, it will need a few tweaks later when all of the styling is refined to better match the designs.
- Tests have been added to the feature specs for admin/rebate_forms/show to ensure the header functions as expected. Similar tests will need to be added to admin/rebate_forms/edit once it is 100% implemented and the existing tests can be removed from pending.

### CODE REVIEW

File: app/helpers/rebate_forms_helper.rb [hunk 1]
```diff
@@ -47,4 +47,8 @@ def rebate_form_boolean_field_to_english(answer)
       answer.to_s.capitalize
     end
   end
+
+  def rebate_form_signature_required?(rebate_form)
+    rebate_form.completed
```
Comment (Br3nda) [inline app/helpers/rebate_forms_helper.rb]: the signature is required if the form is completed? 
isn't it the opposite way around?

File: db/seeds.rb [hunk 2]
```diff
@@ -11,5 +11,5 @@
 
 ['juanvandenanker@gmail.com'].each do |email|
   #email = "#{name}@dia.govt.nz"
-  User.invite!(email: email, roles: [dia_role]) unless User.find_by(email: email)
+  User.invite!(name: 'Ron Weasley', email: email, roles: [dia_role]) unless User.find_by(email: email)
```
Comment (Br3nda) [inline db/seeds.rb]: lol :1st_place_medal:

File: spec/features/admin/batches/index_spec.rb [hunk 3]
```diff
@@ -15,13 +15,6 @@
     end
   end
 
-  context 'anonymous' do
-    it "can't see it" do
-      visit '/admin/batches'
-      expect(page).to have_text('Forgot your password?')
```
Comment (Br3nda) [inline spec/features/admin/batches/index_spec.rb]: why remove this one? is it still a good check that we haven't accidentally left a page open to the public.

File: spec/features/admin/councils/show_spec.rb [hunk 4]
```diff
@@ -15,14 +15,6 @@
     end
   end
 
-  context 'anonymous' do
```
Comment (Br3nda) [inline spec/features/admin/councils/show_spec.rb]: same with these -- we had a mistake once where some pages were correctly blocking anonymous users -- these tests can save us.

File: spec/features/admin/rebate_forms/show_spec.rb [hunk 5]
```diff
@@ -3,10 +3,59 @@
...
+          click_link('reload')
+          expect(page).to have_http_status :ok
+        end
+      end
+
+      context 'when the edit button is clicked' do
+        it 'goes to the right place' do
+          visit "/admin/rebate_forms/#{rebate_form.id}"
+          click_link('edit')
+          expect(page).to_not have_text('EDIT')
```
Comment (Br3nda) [inline spec/features/admin/rebate_forms/show_spec.rb]: can we look for more things on the page here? so we know it actually went to the right place.

File: app/helpers/rebate_forms_helper.rb (hunk 1 above)
Comment (CaraHill) [reply app/helpers/rebate_forms_helper.rb]: @Br3nda  Ok, that wasn't clear in the code. What is the status of the rebate_form if the sgnature is there and it is ready to be processed?

File: spec/features/admin/batches/index_spec.rb (hunk 3 above)
Comment (CaraHill) [reply spec/features/admin/batches/index_spec.rb]: It didn't seem necessary anymore, but if that is the wrong assumption I can put it back.

File: spec/features/admin/councils/show_spec.rb (hunk 4 above)
Comment (CaraHill) [reply spec/features/admin/councils/show_spec.rb]: 👍

File: spec/features/admin/rebate_forms/show_spec.rb (hunk 5 above)
Comment (CaraHill) [reply spec/features/admin/rebate_forms/show_spec.rb]: 👍

File: app/helpers/rebate_forms_helper.rb (hunk 1 above)
Comment (Br3nda) [reply app/helpers/rebate_forms_helper.rb]: What does "required" mean? To me, it means you don't have it yet. Does it mean something else?

If the form is complete, doesn't that mean you've got the signature already? 

Why would you want `signature_required?` to be true when `rebate_form.completed` is true?
If the rebate_form is completed, that means you've got the signatures. It makes it's no longer required.

Comment (Br3nda) [reply app/helpers/rebate_forms_helper.rb]: maybe it should be named `signatures_present?`

Comment (Br3nda) [reply app/helpers/rebate_forms_helper.rb]:
```suggestion
    rebate_form.completed != true
```

File: app/views/admin/rebate_forms/_edit_header.html.haml [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+  = link_to edit_admin_rebate_form_path do
+    =image_tag("blue-reload.svg", class: 'rebate-header-icon')
+
+.rebate-header
+  %h2.rebate-title
+    Customer details
+
+  %h3.rebate-subtitle
+    - if rebate_form_signature_required?(@rebate_form)
+      Signed and ready to process
```
Comment (Br3nda) [inline app/views/admin/rebate_forms/_edit_header.html.haml]:
```suggestion
      Signature required
```

File: app/views/admin/rebate_forms/_edit_header.html.haml [hunk 7]
```diff
@@ -0,0 +1,15 @@
...
+    - else
+      Signature required
```
Comment (Br3nda) [inline app/views/admin/rebate_forms/_edit_header.html.haml]:
```suggestion
 Signed and ready to process
```

File: app/helpers/rebate_forms_helper.rb (hunk 1 above)
Comment (CaraHill) [reply app/helpers/rebate_forms_helper.rb]: I changed it to be called `rebate_form_completed?`

File: spec/features/admin/batches/index_spec.rb (hunk 3 above)
Comment (CaraHill) [reply spec/features/admin/batches/index_spec.rb]: Ok, re added this (and will do all of the others).

**Comment by CaraHill** [comment]
@Br3nda @mischa-s I have addressed all of the comments on this PR and it is ready for another review, thank you!!!

**Pull request merged**